pymeasure/instruments/eurotest/eurotestHPP120256.py

```diff
@@ -0,0 +1,320 @@
+        U, RANGE=3.000kV, VALUE=2.458kV, then voltage will return a tuple
+        (3000.0, 2.458) hence the convenience of the get_process.""",
+        validator=strict_range,
+        values=VOLTAGE_RANGE,
+        get_process=lambda v: (
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[1])[0]),
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[2])[0])
+        )
+    )
+
+    current_limit = Instrument.control(
+        "STATUS,I", "I,%.3fmA",
+        """ A floating point property that represents the output current limit in mAmps setting of the
+        HV Source. This property can be set. When this property acts as get
+        will return a string like this: I, RANGE=5000mA, VALUE=1739mA, then current_limit will
+        return a tuple (5000.0, 1739.0) hence the convenience of the get_process.""",
+        validator=strict_range,
+        values=CURRENT_RANGE,
+        get_process=lambda v: (
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[1])[0]),
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[2])[0])
+        )
+    )
+
+    voltage_ramp = Instrument.control(
+        "STATUS,RAMP", "RAMP,%dV/s",
+        """ A integer property that represents the ramp speed of the output voltage of the
+        HV Source V/s. This property can be set.  When this property acts as get
+        will return a string like this: RAMP, RANGE=3000V/s, VALUE=1000V/s, then voltage_ramp will
+        return a tuple (3000.0, 1000.0) hence the convenience of the get_process.""",
+        validator=strict_range,
+        values=VOLTAGE_RAMP_RANGE,
+        get_process=lambda v: (
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[1])[0]),
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[2])[0])
+        )
+    )
+
+    measure_voltage = Instrument.measurement(
+        "STATUS,MU",
+        """ Measures the actual output voltage of the HV Source in kVolts.
+        This property is a get so will return a string like this:
+        U, RANGE=3.000kV, VALUE=2.458kV, then measure_voltage will
+        return a tuple (3000.0, 2458.0) hence the convenience of the get_process.""",
+        get_process=lambda v: (
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[1])[0]),
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[2])[0])
+        )
+    )
+
+    measure_current = Instrument.measurement(
+        "STATUS,MI",
+        """ Measures the actual output current of the power supply in mAmps.
+        This property is a get so will return a string like this:
+        I, RANGE=5000mA, VALUE=1739mA, then measure_current will
+        return a tuple (5000.0, 1739.0) hence the convenience of the get_process.""",
+        get_process=lambda v: (
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[1])[0]),
+            float(re.findall(r'[-+]?([0-9]*\.[0-9]+|[0-9]+)', v[2])[0])
+        )
+    )
```

Comment:
For me the use of tuples as return values for these properties does not match the idea of using properties.
Why not use more clearer properties like 

- `voltage` (sets the U-setpoint, terunts the actual voltage)
- `current` (sets the I-setpoint, retruns the actual current)
- `current_limit` (..)
- `voltage limit` (..)
- `voltage_ramp` (..)
- `voltage_range` (read-only ?)
- `current_range` (read-only ?)

The doc-strings are intended to give an explanation about the property they describe and do no need to cover details about the implmentation

Reply:
I agree with LongnoseRob in all points.

- If you ask the voltage (inst.voltage), you expect the current voltage and not a tuple.
- The user does not need to know, what string the device sends. You might keep the device responses as a comment for maintainers.

Reply:
Ok. Done in commit. Thanks.